### PR TITLE
Adds tests for _repr_html_ and return None for sparse shape

### PR DESCRIPTION
**Pull request by dtasev**
Adds tests for _repr_html_. It uses BeautifulSoup for parsing the generated HTML so that the output can be tested.

The tests aren't overly sensitive, they mostly check if things are present, but don't necessarily check the full layout.

Makes `.shape` return a ~~Dimension::Sparse~~ `None` value in the position of the shape where the Dimension is sparse. This fixes a workaround for finding where to add the `: Sparse` label.
```
sparse = sc.Variable([Dim.Y, Dim.Z], shape=(3, sc.Dimensions.Sparse))

sparse.dims
[Dim.Y, Dim.Z]

sparse.shape
[3, None]
```

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
> Makes `.shape` return a Dimension::Sparse value in the position of the shape where the Dimension is sparse. This fixes a workaround for finding where to add the `: Sparse` label.

I think we had discussed this before and concluded that it would not be a good idea and break many things. Instead, we should return something like `[2, 3, None]`.

File: python/bind_data_access.h [hunk 1]
```diff
@@ -389,12 +389,12 @@ void bind_data_properties(pybind11::class_<T, Ignored...> &c) {
...
-                            auto shape{std::vector<scipp::index>(
-                                dims.shape().begin(), dims.shape().end())};
+                            py::list list = py::cast(dims.shape());
+
                             if (dims.sparse()) {
-                              shape.emplace_back(Dimensions::Sparse);
+                              list.append(py::none());
                             }
-                            return shape;
+                            return list;
```
Comment (SimonHeybrock) [inline python/bind_data_access.h]: Can we add a unit test for this?

@nvaytet Does this change have any effect on the plotting code?

File: python/tests/table_html/test_to_html_dataset.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pytest
+from bs4 import BeautifulSoup
```
Comment (SimonHeybrock) [inline python/tests/table_html/test_to_html_dataset.py]: Does this need to be installed on Travis?

File: python/bind_data_access.h (hunk 1 above)
Comment (dtasev) [reply python/bind_data_access.h]: Just to note that now both Variables and Dataset will return None on the sparse dim's shape

File: python/tests/table_html/test_to_html_dataset.py (hunk 2 above)
Comment (dtasev) [reply python/tests/table_html/test_to_html_dataset.py]: Yes. I thought I had it to `requirements.txt` but I can't see it, so I'll do that

File: python/bind_data_access.h [hunk 3]
```diff
@@ -390,8 +390,12 @@ void bind_data_properties(pybind11::class_<T, Ignored...> &c) {
   c.def_property_readonly("shape",
                           [](const T &self) {
                             const auto &dims = self.dims();
-                            return std::vector<scipp::index>(
-                                dims.shape().begin(), dims.shape().end());
+                            py::list list = py::cast(dims.shape());
```
Comment (SimonHeybrock) [inline python/bind_data_access.h]: There are some issues with Apple clang. Could this be the reason? It am not sure what `py::cast` does here? Can we use iterators or some other way to setup the list?

Comment (OwenArnold) [reply python/bind_data_access.h]: Fix on https://github.com/scipp/scipp/commits/dtasev-table_html_testing or can provide a patch separately.

Comment (SimonHeybrock) [reply python/bind_data_access.h]: Can you create a PR into this branch?

**Comment by dtasev** [comment]
Everything is passing now.

**Comment by SimonHeybrock** [comment]
@dtasev Have you tried building the docs? There are some Python changes that might break something.

**Pull request merged**